src/__tests__/reporter/reporter.spec.ts

```diff
@@ -0,0 +1,218 @@
+      expect(reporter['client'].startTestItem).toHaveBeenCalledTimes(1);
+      expect(reporter['client'].startTestItem).toHaveBeenCalledWith(startSuiteObj, 'tempLaunchId');
+    });
+
+    test('reporter.suiteIds should be updated ', () => {
+      expect(reporter['suiteIds']).toEqual(['tempTestItemId']);
+    });
+  });
+
+  describe('start report test item', () => {
+    const reporter = setupReporter(['launchId', 'suiteIds']);
+    const testMeta = {
+      description: 'test_description',
+    };
+    const startTestObj: StartTestItemRQ = {
+      name: testName,
+      type: TEST_ITEM_TYPES.STEP,
+      description: testMeta.description,
+    };
+
+    reporter.reportTestStart(testName, testMeta);
+
+    test('client.startTestItem should be called with corresponding params', () => {
+      expect(reporter['client'].startTestItem).toHaveBeenCalledTimes(1);
+      expect(reporter['client'].startTestItem).toHaveBeenCalledWith(
+        startTestObj,
+        'tempLaunchId',
+        'tempTestItemId',
+      );
+    });
+
+    test('reporter.testItems should be updated', () => {
+      expect(reporter['testItems']).toEqual([{ name: testName, id: 'tempTestItemId' }]);
+    });
+  });
+
+  describe('finish report test item', () => {
+    const testRunInfo: any = {
+      errs: [],
+      skipped: false,
+    };
+
+    describe('test with PASSED status', () => {
+      const reporter = setupReporter(['launchId', 'suiteIds', 'testItems']);
+
+      reporter.reportTestDone(testName, testRunInfo);
+
+      test('client.finishTestItem should be called with corresponding params', () => {
+        expect(reporter['client'].finishTestItem).toHaveBeenCalledTimes(1);
+        expect(reporter['client'].finishTestItem).toHaveBeenCalledWith('tempTestItemId', {
+          status: STATUSES.PASSED,
+        });
+      });
+
+      test('reporter.testItems should be []', () => {
+        expect(reporter['testItems']).toEqual([]);
+      });
+    });
+
+    describe('test with SKIPPED status', () => {
+      const reporter = setupReporter(['launchId', 'suiteIds', 'testItems']);
+      testRunInfo.skipped = true;
+
+      reporter.reportTestDone(testName, testRunInfo);
+
+      test('client.finishTestItem should be called with corresponding params', () => {
+        expect(reporter['client'].finishTestItem).toHaveBeenCalledTimes(1);
+        expect(reporter['client'].finishTestItem).toHaveBeenCalledWith('tempTestItemId', {
+          status: STATUSES.SKIPPED,
+        });
+      });
+
+      test('reporter.testItems should be []', () => {
+        expect(reporter['testItems']).toEqual([]);
+      });
+    });
+
+    describe('test with FAILED status', () => {
+      const reporter = setupReporter(['launchId', 'suiteIds', 'testItems']);
+      testRunInfo.skipped = false;
+      testRunInfo.errs = ['401 unauthorized'];
+      reporter.sendLogsOnFail = jest.fn((errors, testItemId) => ({ errors, testItemId }));
```

Comment:
Why not just mock this function implementation?
F.e.
```javascript
jest.spyOn(reporter, 'sendLogsOnFail').mockImplementation(() => {});
```
It will simplify parameters cheching for [these](https://github.com/reportportal/agent-js-testcafe/pull/6/files#diff-0571ce57a1ca3a45fea6b987d7592d0a5712399e203134d2f2c19df3ee66013eR184-R188) lines.